### PR TITLE
added sigmoid_focal_loss  to paddle frontend and refactored mse_loss in paddle frontend

### LINKED ISSUE #19759: sigmoid_focal_loss

### LINKED ISSUE #22217: square_error_cost

### PR DESCRIPTION
**Pull request by zeus2x7**
close #19759
subtask #15053
Refactored mse_loss in paddle frontend to pass all the tests

### CODE REVIEW

**Comment by zeus2x7** [comment]
hi @sagewhocodes  can you please review the pr ?

**Comment by zeus2x7** [comment]
@sagewhocodes  can you please review the pr ?

**Comment by zeus2x7** [comment]
Hi @MuhammedAshraf2020 can you please review the pr?

**Comment by zeus2x7** [comment]
Hi @MuhammedAshraf2020 can you please review the pr?

**Comment by zeus2x7** [comment]
@MuhammedAshraf2020  it would be really helpful if you could review the pr

**Comment by zeus2x7** [comment]
@MuhammedAshraf2020  it would be really helpful if you could review the pr

**Comment by MuhammedAshraf2020** [comment]
Hi @zeus2x7 
Thanks for the PR. There seems to be a few small issues that I pointed out below.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_loss.py [hunk 1]
```diff
@@ -468,6 +468,87 @@ def test_paddle_nll_loss(
...
+    fn_tree="paddle.nn.functional.sigmoid_focal_loss",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+        num_arrays=1,
+        shared_dtype=False,
+        min_num_dims=3,
+        min_dim_size=3,
+        max_num_dims=3,
+        max_dim_size=3,
+    ),
```
Comment (MuhammedAshraf2020) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_loss.py:481]: This doesn't cover the entire input space

**Comment by zeus2x7** [comment]
> Hi @zeus2x7 Thanks for the PR. There seems to be a few small issues that I pointed out below.

Where are the issues listed ?

**Review by MuhammedAshraf2020 — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_loss.py (hunk 1 above)
Comment (zeus2x7) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_loss.py:481]: @MuhammedAshraf2020  i had to fix the dimensions of the input to match with those of the labels as their dtype is different.

**Comment by zeus2x7** [comment]
hi @MuhammedAshraf2020  can you please review it now?

**Comment by MuhammedAshraf2020** [comment]
Great, Thanks

**Review by MuhammedAshraf2020 — APPROVED** [review]
